diff -rNu crash-git.orig//extensions/sial.c crash-git/extensions/sial.c
--- crash-git.orig//extensions/sial.c	2014-09-18 00:57:19.489890649 +0000
+++ crash-git/extensions/sial.c	2014-09-18 01:11:43.809892227 +0000
@@ -512,6 +512,7 @@
 	{"LINUX_2_2_X",		"(((LINUX_RELEASE) & 0xffff00) == 0x020200)"},
 	{"LINUX_2_4_X",		"(((LINUX_RELEASE) & 0xffff00) == 0x020400)"},
 	{"LINUX_2_6_X",		"(((LINUX_RELEASE) & 0xffff00) == 0x020600)"},
+	{"LINUX_3_X_X",         "(((LINUX_RELEASE) & 0xff0000) == 0x030000)"},
 #ifdef i386
 	{"i386",         "1"},
 	{"__i386",       "1"},
@@ -621,12 +622,17 @@
                     if(tok) tok=strtok(NULL, " \t");
                     if(tok) tok=strtok(NULL, " \t");
                     if(tok) {
-                        int two, major, minor, ret;
-                        ret = sscanf(tok, "%d.%d.%d-", &two, &major, &minor);
-                        if( ret == 3) {
-                            sprintf(banner, "0x%02x%02x%02x", two, major, minor);
-		            dt=sial_add_def(dt, sial_strdup("LINUX_RELEASE"), sial_strdup(banner));
-                            sial_msg("Core LINUX_RELEASE == '%s'\n", tok);
+                        int version, patchlevel, sublevel, ret;
+                        ret = sscanf(tok, "%d.%d.%d-", &version, &patchlevel, &sublevel);
+			switch (ret) {
+			case 2:
+			    sublevel = 0;
+			case 3:
+			    sprintf(banner, "0x%02x%02x%02x", version, patchlevel, sublevel);
+ 		            dt=sial_add_def(dt, sial_strdup("LINUX_RELEASE"), sial_strdup(banner));
+                             sial_msg("Core LINUX_RELEASE == '%s'\n", tok);
+			default:
+			    break;
                         }
                     }
                 }
diff -rNu crash-git.orig//kernel.c crash-git/kernel.c
--- crash-git.orig//kernel.c	2014-09-18 00:57:19.489890649 +0000
+++ crash-git/kernel.c	2014-09-18 01:09:06.301903490 +0000
@@ -1016,7 +1016,8 @@
 	found = FALSE;
 	sprintf(buffer3, "(unknown)");
         while (fgets(buffer, BUFSIZE-1, pipe)) {
-		if (!strstr(buffer, "Linux version 2."))
+		if (!strstr(buffer, "Linux version 2.") &&
+		    !strstr(buffer, "Linux version 3."))
 			continue;
 
                 if (strstr(buffer, kt->proc_version)) {
@@ -2159,7 +2160,7 @@
 	ulong *up;
 	char buf[BUFSIZE];
 	ulong eip, esp;
-	struct bt_info btsave;
+	struct bt_info btsave = { 0 };
 
 	if (bt->flags & BT_RAW) {
 		if (bt->hp && bt->hp->esp)
