--- tcpdump-4.1.1-1.org/configure.in	2012-08-22 15:37:18.000000000 +0000
+++ tcpdump-4.1.1-1/configure.in	2012-08-22 15:37:44.000000000 +0000
@@ -72,9 +72,9 @@
  			sed -n -e '$s/.* //' -e '$s/\..*//p'`)
  	fi
 	AC_MSG_RESULT($ac_cv_linux_vers)
- 	if test $ac_cv_linux_vers = unknown ; then
- 		AC_MSG_ERROR(cannot determine linux version when cross-compiling)
- 	fi
+# 	if test $ac_cv_linux_vers = unknown ; then
+#		AC_MSG_ERROR(cannot determine linux version when cross-compiling)
+# 	fi
 	if test $ac_cv_linux_vers -lt 2 ; then
 		AC_MSG_ERROR(version 2 or higher required; see the INSTALL doc for more info)
 	fi
--- tcpdump-4.1.1-1.org/configure	2012-08-22 15:42:30.000000000 +0000
+++ tcpdump-4.1.1-1/configure	2012-08-22 15:42:59.000000000 +0000
@@ -4409,11 +4409,11 @@
  	fi
 	{ echo "$as_me:$LINENO: result: $ac_cv_linux_vers" >&5
 echo "${ECHO_T}$ac_cv_linux_vers" >&6; }
- 	if test $ac_cv_linux_vers = unknown ; then
- 		{ { echo "$as_me:$LINENO: error: cannot determine linux version when cross-compiling" >&5
-echo "$as_me: error: cannot determine linux version when cross-compiling" >&2;}
-   { (exit 1); exit 1; }; }
- 	fi
+# 	if test $ac_cv_linux_vers = unknown ; then
+# 		{ { echo "$as_me:$LINENO: error: cannot determine linux version when cross-compiling" >&5
+#echo "$as_me: error: cannot determine linux version when cross-compiling" >&2;}
+#   { (exit 1); exit 1; }; }
+# 	fi
 	if test $ac_cv_linux_vers -lt 2 ; then
 		{ { echo "$as_me:$LINENO: error: version 2 or higher required; see the INSTALL doc for more info" >&5
 echo "$as_me: error: version 2 or higher required; see the INSTALL doc for more info" >&2;}
