diff -Naur old/configure.in heartbeat-git/configure.in
--- old/configure.in	2014-12-25 16:04:42.000000000 +0700
+++ heartbeat-git/configure.in	2014-12-25 16:28:39.000000000 +0700
@@ -137,7 +137,7 @@
 	  printf "#include <stdio.h>\n" > ${Cfile}.c
 	  printf "#include <%s>\n" $1 >> ${Cfile}.c
 	  printf "int main(int argc, char **argv) { printf(\"%%s\", %s); return 0; }\n" $2 >> ${Cfile}.c
-	  $CC $CFLAGS ${Cfile}.c -o ${Cfile}
+	  gcc $CFLAGS ${Cfile}.c -o ${Cfile}
 	  value=`${Cfile}`
 	  AC_MSG_RESULT($value)
 	  printf $value
