diff -Naur heartbeat-3.0.3-2.org//configure.in heartbeat-3.0.3-2/configure.in
--- heartbeat-3.0.3-2.org//configure.in	2012-09-05 09:34:06.000000000 +0000
+++ heartbeat-3.0.3-2/configure.in	2012-09-05 09:34:52.000000000 +0000
@@ -2222,10 +2222,10 @@
 
 dnl ************************************************************************
 dnl * Check for (debian style?) $sysconfdir/pam.d/common-{account,auth}
-AC_CHECK_FILE($sysconfdir/pam.d/common-account)
-AC_CHECK_FILE($sysconfdir/pam.d/common-auth)
-AM_CONDITIONAL(USE_PAM_COMMON, test -f $sysconfdir/pam.d/common-account -a \
-				    -f /etc/pam.d/common-auth)
+#AC_CHECK_FILE($sysconfdir/pam.d/common-account)
+#AC_CHECK_FILE($sysconfdir/pam.d/common-auth)
+#AM_CONDITIONAL(USE_PAM_COMMON, test -f $sysconfdir/pam.d/common-account -a \
+#				    -f /etc/pam.d/common-auth)
 
 dnl GNUTLS library: Attempt to determine by 'libgnutls-config' program.
 dnl If no 'libgnutls-config', try traditional autoconf means.
