diff -Naur policycoreutils-2.0.82-3.org//run_init/Makefile policycoreutils-2.0.82-3/run_init/Makefile
--- policycoreutils-2.0.82-3.org//run_init/Makefile	2012-08-29 17:10:22.000000000 +0000
+++ policycoreutils-2.0.82-3/run_init/Makefile	2012-08-29 17:11:03.000000000 +0000
@@ -10,7 +10,7 @@
 
 CFLAGS ?= -Werror -Wall -W
 override CFLAGS += -I$(PREFIX)/include -DUSE_NLS -DLOCALEDIR="\"$(LOCALEDIR)\"" -DPACKAGE="\"policycoreutils\""
-LDLIBS += -lselinux -L$(PREFIX)/lib
+LDLIBS += -lselinux -L$(setlib) -I$(setinc) 
 ifeq (${PAMH}, /usr/include/security/pam_appl.h)
 	override CFLAGS += -DUSE_PAM
 	LDLIBS += -lpam -lpam_misc
diff -Naur policycoreutils-2.0.82-3.org//run_init/Makefile policycoreutils-2.0.82-3/run_init/Makefile
--- policycoreutils-2.0.82-3.org//run_init/Makefile	2012-08-30 15:26:37.000000000 +0000
+++ policycoreutils-2.0.82-3/run_init/Makefile	2012-08-30 15:26:59.000000000 +0000
@@ -9,7 +9,7 @@
 AUDITH = $(shell ls /usr/include/libaudit.h 2>/dev/null)
 
 CFLAGS ?= -Werror -Wall -W
-override CFLAGS += -I$(PREFIX)/include -DUSE_NLS -DLOCALEDIR="\"$(LOCALEDIR)\"" -DPACKAGE="\"policycoreutils\""
+override CFLAGS += -DUSE_NLS -DLOCALEDIR="\"$(LOCALEDIR)\"" -DPACKAGE="\"policycoreutils\""
 LDLIBS += -lselinux -L$(setlib) -I$(setinc) 
 ifeq (${PAMH}, /usr/include/security/pam_appl.h)
 	override CFLAGS += -DUSE_PAM
