diff -Naur a/testcases/kernel/timers/clock_settime/clock_settime03.c b/testcases/kernel/timers/clock_settime/clock_settime03.c
--- a/testcases/kernel/timers/clock_settime/clock_settime03.c	2012-02-27 07:16:37.060211645 +0000
+++ b/testcases/kernel/timers/clock_settime/clock_settime03.c	2012-02-27 07:27:36.100209029 +0000
@@ -129,7 +129,7 @@
 	/* PROCESS_CPUTIME_ID & THREAD_CPUTIME_ID are not supported on
 	 * kernel versions lower than 2.6.12
 	 */
-	if((tst_kvercmp(2, 6, 12)) < 0) {
+	if((tst_kvercmp(2, 6, 12)) < 0 || (tst_kvercmp(2, 6, 38)) >= 0) {
 		testcase[7] = EINVAL;
 		testcase[8] = EINVAL;
 	} else {
