diff -Naur a/runtest/syscalls b/runtest/syscalls
--- a/runtest/syscalls	2012-02-09 16:37:44.000000000 +0900
+++ b/runtest/syscalls	2012-02-09 16:43:33.000000000 +0900
@@ -464,7 +464,6 @@
 lchown02 lchown02
 lchown02_16 lchown02_16
 
-libevent01 export LTPROOT; $LTPROOT/testcases/bin/libevent/run_libevent.sh
 link01 symlink01 -T link01
 link02 link02
 link03 link03
diff -Naur a/testcases/kernel/syscalls/Makefile b/testcases/kernel/syscalls/Makefile
--- a/testcases/kernel/syscalls/Makefile	2012-02-09 16:37:45.000000000 +0900
+++ b/testcases/kernel/syscalls/Makefile	2012-02-09 16:43:36.000000000 +0900
@@ -24,53 +24,9 @@
 
 include	$(top_srcdir)/include/mk/env_pre.mk
 
-# libevent requires some massaging to get this to work.
-FILTER_OUT_DIRS		:= libevent
 # pcllib is busted as all heck because of libtool
 FILTER_OUT_DIRS		+= pcllib
 
-ifeq ($(wildcard $(abs_srcdir)/libevent/test/Makefile),)
-$(warning libevent/test/Makefile does not exist; skipping to avoid compile errors)
-$(warning Please run "make autotools" and "configure" at the top-level LTP)
-$(warning directory if you want to run this test)
-else
-
-LIBEVENT		:= libevent/lib/libevent.a
-
-CLEAN_DEPS		:= libevent-clean
-
-CLEAN_TARGETS		:= $(LIBEVENT)
-
-INSTALL_DEPS		:= libevent-install
-
-INSTALL_TARGETS		:= libevent/run_libevent.sh libevent/test/test-libevent.sh
-
-LIBEVENT_APPS		:= regress test-eof test-init test-time test-weof
-
-TEST_APPS		:= $(addprefix libevent/test/,$(LIBEVENT_APPS))
-
-MAKE_DEPS		:= libevent-all
-
-MAKE_TARGETS		+= $(TEST_APPS)
-
-libevent libevent/test: %:
-	mkdir -p "$@"
-
-$(TEST_APPS): CPPFLAGS += -I$(abs_srcdir)/libevent
-$(TEST_APPS): LDFLAGS += -L$(abs_builddir)/libevent
-$(TEST_APPS): LDLIBS += -levent
-
-$(LIBEVENT): $(abs_srcdir)/libevent/Makefile libevent
-	$(MAKE) -C libevent -f "$(firstword $^)" libevent.a
-
-libevent-all: $(abs_srcdir)/libevent/test/Makefile | $(LIBEVENT) libevent/test
-	$(MAKE) -C libevent/test -f "$(firstword $^)" $(LIBEVENT_APPS)
-
-libevent-clean libevent-install: libevent-%: $(abs_srcdir)/libevent/test/Makefile | libevent/test
-	$(MAKE) -C libevent/test -f "$(firstword $^)" $*
-
-endif
-
 ifeq ($(UCLINUX),1)
 FILTER_OUT_DIRS	+= capget capset chmod chown clone fork getcontext llseek \
 		   nftw profil remap_file_pages
