diff -Naur a/testcases/kernel/syscalls/sockioctl/sockioctl01.c b/testcases/kernel/syscalls/sockioctl/sockioctl01.c
--- a/testcases/kernel/syscalls/sockioctl/sockioctl01.c	2012-02-28 06:12:02.188137374 +0000
+++ b/testcases/kernel/syscalls/sockioctl/sockioctl01.c	2012-02-28 06:13:28.832211959 +0000
@@ -228,6 +228,11 @@
 			 strerror(errno));
 	}
 	sinlen = sizeof(fsin1);
+
+	if (strncmp(tdat[testno].desc, "ATMARK on UDP", 14) == 0) {
+		if ((tst_kvercmp(2, 6, 39)) >= 0)
+			tdat[testno].experrno = ENOTTY;
+	}
 }
 
 void setup2(void)
