diff -Naur a/src/cap-ng.c libcap-ng-git/src/cap-ng.c
--- a/src/cap-ng.c	2014-09-18 07:57:55.000000000 +0000
+++ libcap-ng-git/src/cap-ng.c	2014-09-18 07:59:01.000000000 +0000
@@ -90,6 +90,18 @@
 #define _LINUX_CAPABILITY_VERSION_3 0x20080522
 #endif
 
+// This public API went private in the 2.6.36 kernel - hope it never changes
+#ifndef XATTR_CAPS_SUFFIX
+#define XATTR_CAPS_SUFFIX "capability"
+#endif
+#ifndef XATTR_SECURITY_PREFIX
+#define XATTR_SECURITY_PREFIX "security."
+#endif
+#ifndef XATTR_NAME_CAPS
+#define XATTR_NAME_CAPS XATTR_SECURITY_PREFIX XATTR_CAPS_SUFFIX
+#endif
+
+
 /* Child processes can't get caps back */
 #ifndef SECURE_NOROOT
 #define SECURE_NOROOT                   0
