diff -Naur cluster-glue-1.0.6-1.orig/lib/plugins/stonith/Makefile.am cluster-glue-1.0.6-1/lib/plugins/stonith/Makefile.am
--- cluster-glue-1.0.6-1.orig/lib/plugins/stonith/Makefile.am	2012-08-30 17:29:13.000000000 +0000
+++ cluster-glue-1.0.6-1/lib/plugins/stonith/Makefile.am	2012-08-30 17:29:48.000000000 +0000
@@ -148,7 +148,7 @@
 
 drac3_la_SOURCES	= drac3.c drac3_command.c drac3_command.h drac3_hash.c drac3_hash.h	$(INCFILES)
 drac3_la_LDFLAGS	= -export-dynamic -module -avoid-version
-drac3_la_LIBADD		= -lcurl -lxml2 $(GLIBLIB)
+drac3_la_LIBADD		= -lcurl-gnutls -lxml2 $(GLIBLIB)
 
 external_la_SOURCES	= external.c	$(INCFILES)
 external_la_LDFLAGS	= -export-dynamic -module -avoid-version
diff -Naur cluster-glue-1.0.6-1.orig/doc/Makefile.am cluster-glue-1.0.6-1/doc/Makefile.am
--- cluster-glue-1.0.6-1.orig/doc/Makefile.am	2012-08-30 18:20:14.000000000 +0000
+++ cluster-glue-1.0.6-1/doc/Makefile.am	2012-08-30 18:22:01.000000000 +0000
@@ -28,7 +28,7 @@
 man_MANS		=
 
 if BUILD_DOC
-man_MANS		+= hb_report.8 ha_logd.8 ha_logger.1 stonith.8 meatclient.8
+#man_MANS		+= hb_report.8 ha_logd.8 ha_logger.1 stonith.8 meatclient.8
 endif
 
 EXTRA_DIST		= $(man_MANS)
